src/lib/device/device_manager.cpp

```diff
@@ -217,20 +218,21 @@ shared< VirtualDev > DeviceManager::create_vdev(vdev_parameters&& vparam) {
 
         // Either num_chunks or chunk_size can be provided and we calculate the other.
         if (vparam.num_chunks != 0) {
-            vparam.vdev_size = sisl::round_up(vparam.vdev_size, vparam.num_chunks * vparam.blk_size);
-            if (input_vdev_size != vparam.vdev_size) {
-                LOGINFO(
-                    "{} Virtual device is attempted to be created with size={}, it needs to be rounded to new_size={}",
-                    vparam.vdev_name, in_bytes(input_vdev_size), in_bytes(vparam.vdev_size));
-            }
-
             auto input_num_chunks = vparam.num_chunks;
             vparam.num_chunks = std::min(vparam.num_chunks, max_num_chunks);
             if (input_num_chunks != vparam.num_chunks) {
                 LOGINFO("{} Virtual device is attempted to be created with num_chunks={}, it needs to be adjust to "
                         "new_num_chunks={}",
                         vparam.vdev_name, in_bytes(input_num_chunks), in_bytes(vparam.num_chunks));
             }
+
+            vparam.vdev_size = sisl::round_up(vparam.vdev_size, vparam.num_chunks * vparam.blk_size);
+            if (input_vdev_size != vparam.vdev_size) {
+                LOGINFO(
+                    "{} Virtual device is attempted to be created with size={}, it needs to be rounded to new_size={}",
+                    vparam.vdev_name, in_bytes(input_vdev_size), in_bytes(vparam.vdev_size));
+            }
+
```

Comment:
we should first compute the final vparam.num_chunks, then compute vparam.vdev_size according to the final vparam.num_chunks, so that we can get a aligned chunk size.

otherwise , we will first get a vdev_size from the num_chunks of direct input vparam.num_chunks, which might lead to unalignment